filters/auth/jwt_metrics_test.go

```diff
@@ -223,7 +225,44 @@ func TestJwtMetrics(t *testing.T) {
 			`,
 			request: &http.Request{Method: "GET", Host: "foo.test",
 				Header: http.Header{"Authorization": []string{
-					"Bearer foobarbaz",
+					validAuthHeader,
+				}},
+			},
+			status: http.StatusOK,
+			expected: map[string]int64{
+				"jwtMetrics.custom.GET.foo_test.200.invalid-token": 1,
+			},
+			expectedTag: "invalid-token",
+		},
+		{
+			name:     "no metrics when host matches opted-out domain",
+			filters:  `jwtMetrics("{issuers: [foo, bar], optOutHosts: [ '^.+[.]domain[.]test$', '^exact[.]test$' ]}")`,
+			request:  &http.Request{Method: "GET", Host: "foo.domain.test"},
+			status:   http.StatusOK,
+			expected: map[string]int64{},
+		},
+		{
+			name:     "no metrics when second level host matches opted-out domain",
+			filters:  `jwtMetrics("{issuers: [foo, bar], optOutHosts: [ '^.+[.]domain[.]test$', '^exact[.]test$' ]}")`,
+			request:  &http.Request{Method: "GET", Host: "foo.bar.domain.test"},
+			status:   http.StatusOK,
+			expected: map[string]int64{},
+		},
+		{
+			name:     "no metrics when host matches opted-out host exactly",
+			filters:  `jwtMetrics("{issuers: [foo, bar], optOutHosts: [ '^.+[.]domain[.]test$', '^exact[.]test$' ]}")`,
+			request:  &http.Request{Method: "GET", Host: "exact.test"},
```

Comment:
Should we test the same filter, but with "not.exact.test" and/or "notexact.test" host?

Reply:
There is "counts invalid-token when host does not match" test case for that.